lib/external-grading-socket.js

```diff
@@ -0,0 +1,54 @@
+var ERR = require('async-stacktrace');
+var _ = require('lodash');
+var async = require('async');
+var child_process = require('child_process');
+
+var logger = require('./logger');
+var socketServer = require('./socket-server');
+var sqldb = require('./sqldb');
+var sqlLoader = require('./sql-loader');
+
+var sql = sqlLoader.loadSqlEquiv(__filename);
+
+module.exports = {}
+
+// This module MUST be initialized after socket-server
+module.exports.init = function(callback) {
+    this.namespace = socketServer.io.of('/external-grading');
+    this.namespace.on('connection', this.connection);
+
+    callback(null);
+};
+
+module.exports.connection = function(socket) {
+    socket.on('getStatus', (msg, callback) => {
+        if (!_.has(msg, 'job_id')) {
```

Comment:
Can we rename `job_id` to `grading_job_id` everywhere to avoid confusion with the `job_id` in `server-jobs.js`? I'd be fine with also renaming the `jobs` table to `server_jobs` and using `server_job_id` everywhere if you want.